src/main/java/org/jabref/gui/undo/RedoAction.java

```diff
@@ -22,9 +21,10 @@ public RedoAction(Supplier<LibraryTab> tabSupplier, DialogService dialogService,
         this.tabSupplier = tabSupplier;
         this.dialogService = dialogService;
 
-        // TODO: Rework the UndoManager to something like the following, if it had a property.
-        //  this.executable.bind(frame.getCurrentBasePanel().getUndoManager().canUndo())
-        this.executable.bind(ActionHelper.needsDatabase(stateManager));
+        stateManager.activeTabProperty().addListener((observable, oldValue, activeLibraryTab) -> {
```

Comment:
This will add a listener after each tab change.  The old listener should be removed. Otherwise, memory consumption will increase IMHO.